File: app/src/main/java/me/chrislane/accudrop/db/UserDao.java
package me.chrislane.accudrop.db;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Query;

import java.util.UUID;

@Dao
public interface UserDao {

    @Query("SELECT * FROM user")
    LiveData<User> findAllUsers();

    @Query("SELECT name FROM user " +
            "WHERE uuid = :uuid")
    LiveData<String> findNameFromUUID(UUID uuid);
}

File: app/src/main/java/me/chrislane/accudrop/db/UserDao.kt
package me.chrislane.accudrop.db

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import java.util.*

@Dao
interface UserDao {

  @Query("SELECT * FROM user")
  fun findAllUsers(): LiveData<User>

  @Query("SELECT * FROM user WHERE uuid = :uuid")
  fun getUserById(uuid: UUID): User

  @Insert
  fun insertUser(user: User)

}
